util/headerreader/header_reader.go

```diff
@@ -434,6 +434,10 @@ func (s *HeaderReader) Client() arbutil.L1Interface {
 	return s.client
 }
 
+func (s *HeaderReader) UseFinalityData() bool {
```

Comment:
Unless I'm missing something.. Seems like callopts in both l1Validator and Staker are initialized without params.
What if HeaderReader instead implemented a "getcallopts" function, that would return callopts which request the appropriate block?
This could simplify some things.

Reply:
could have a param to such function that optionally ignores the treatfinality if in some cases you always want latest.

Reply:
I think we'll just want to wait until the new challenge protocol replaces a lot of this code in this case